parsec/core/mountpoint/winfsp_runner.py

```diff
@@ -20,80 +22,56 @@
 logger = get_logger()
 
 
-async def cleanup_broken_links(path: trio.Path) -> None:
-    def target():
-        for child in sync_path.iterdir():
-            try:
-                if not child.exists():
-                    child.unlink()
-            except OSError:
-                pass
-
-    # This should be migrated to trio.Path API once issue #1308 is fixed.
-    sync_path = Path(str(path))
-    await trio.to_thread.run_sync(target)
-
-
-async def is_path_available(path: trio.Path) -> bool:
-    # The path already exists
-    if await path.exists():
-        return False
-
-    # The path is a broken link
-    try:
-        await path.lstat()
-        return True
-    except OSError:
-        pass
-
-    # The path is available
-    return True
-
-
-async def _bootstrap_mountpoint(base_mountpoint_path: Path, workspace_name) -> Path:
-    # Mountpoint can be a drive letter, in such case nothing to do
-    if str(base_mountpoint_path) == base_mountpoint_path.drive:
-        return
-
-    # On Windows, only mountpoint's parent must exists
-    trio_base_mountpoint_path = trio.Path(base_mountpoint_path)
-    await trio_base_mountpoint_path.mkdir(exist_ok=True, parents=True)
+BASE = "PQRSTUVWXYZHIJKLMNO"
+assert len(BASE) == 19
 
-    # Clean up broken links in base directory
-    await cleanup_broken_links(trio_base_mountpoint_path)
 
-    # Find a suitable path where to mount the workspace. The check we are doing
-    # here are not atomic (and the mount operation is not itself atomic anyway),
-    # hence there is still edgecases where the mount can crash due to concurrent
-    # changes on the mountpoint path
-    for tentative in count(1):
-        if tentative == 1:
-            dirname = workspace_name
-        else:
-            dirname = f"{workspace_name} ({tentative})"
+def candidates(index, length, nearest=5):
```

Comment:
Better naming ? (for instance `_workspace_id_to_drive_letter`)
Adding typing would be good too, and we should replace the str return type by a `List[str]` (better to inform we return multiple drive letters)

And document this non-trivial code ?

Reply:
:+1: